crypto/ecies.go

```diff
@@ -0,0 +1,32 @@
+package crypto
+
+import (
+	"crypto/ecdsa"
+
+	ecies "github.com/ecies/go"
+)
+
+// EciesDecrypt decrypts the `cipherText` using the Elliptic Curve Integrated Encryption Scheme
+func EciesDecrypt(privateKey *ecdsa.PrivateKey, cipherText []byte) ([]byte, error) {
```

Comment:
Can you make this implement the `crypto.Decrypter` interface?

Reply:
I did this on purpose as `EciesEncrypt`/`EciesDecrypt` are generic functions to encrypt/decrypt using ECIES with the standard Go structures for `crypto/ecdsa` private and public keys. The `store.Decrypt` method is the actual implementation of `crypto.Decrypter`

Reply:
I mean the golang `crypto.Decrypter` which has the following signature `Decrypt(rand io.Reader, msg []byte, opts crypto.DecrypterOpts) (plaintext []byte, err error)`

>// Decrypter is an interface for an opaque private key that can be used for
// asymmetric decryption operations. An example would be an RSA key
// kept in a hardware module.

Reply:
Ah okay, none of the ECIES libraries are compatible with `crypto.Decrypter` unfortunately (they require access to the actual private key instead)

Reply:
We could implement `crypto.Decrypter` for our crypto store but it doesn't really matter for the `EciesDecrypt` function as that's only an helper for ECIES encrypt/decrypt.